comp/metadata/inventoryotel/inventoryotelimpl/inventoryotel.go

```diff
@@ -0,0 +1,270 @@
+	"github.com/DataDog/datadog-agent/pkg/serializer/marshaler"
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+	"github.com/DataDog/datadog-agent/pkg/util/hostname"
+	"github.com/DataDog/datadog-agent/pkg/util/scrubber"
+	"github.com/DataDog/datadog-agent/pkg/util/uuid"
+)
+
+// Module defines the fx options for this component.
+func Module() fxutil.Module {
+	return fxutil.Component(
+		fx.Provide(newInventoryOtelProvider))
+}
+
+type otelMetadata = map[string]interface{}
+
+// Payload handles the JSON unmarshalling of the metadata payload
+type Payload struct {
+	Hostname  string       `json:"hostname"`
+	Timestamp int64        `json:"timestamp"`
+	Metadata  otelMetadata `json:"otel_metadata"`
+	UUID      string       `json:"uuid"`
+}
+
+// MarshalJSON serialization a Payload to JSON
+func (p *Payload) MarshalJSON() ([]byte, error) {
+	type PayloadAlias Payload
+	return json.Marshal((*PayloadAlias)(p))
+}
+
+// SplitPayload implements marshaler.AbstractMarshaler#SplitPayload.
+//
+// In this case, the payload can't be split any further.
+func (p *Payload) SplitPayload(_ int) ([]marshaler.AbstractMarshaler, error) {
+	return nil, fmt.Errorf("could not split inventories agent payload any more, payload is too big for intake")
+}
+
+type inventoryotel struct {
+	util.InventoryPayload
+
+	conf      config.Component
+	log       log.Component
+	m         sync.Mutex
+	data      otelMetadata
+	hostname  string
+	authToken authtoken.Component
+	f         *freshConfig
+}
+
+type dependencies struct {
+	fx.In
+
+	Log        log.Component
+	Config     config.Component
+	Serializer serializer.MetricSerializer
+	AuthToken  authtoken.Component
+}
+
+type provides struct {
+	fx.Out
+
+	Comp                 iointerface.Component
+	Provider             runnerimpl.Provider
+	FlareProvider        flaretypes.Provider
+	StatusHeaderProvider status.HeaderInformationProvider
+	Endpoint             api.AgentEndpointProvider
+}
+
+func newInventoryOtelProvider(deps dependencies) provides {
+	hname, _ := hostname.Get(context.Background())
+	i := &inventoryotel{
+		conf:      deps.Config,
+		log:       deps.Log,
+		hostname:  hname,
+		data:      make(otelMetadata),
+		authToken: deps.AuthToken,
+	}
+
+	getter := i.fetchRemoteOtelConfig
+	if i.conf.GetBool("otel.submit_dummy_inventories_metadata") {
+		getter = i.fetchDummyOtelConfig
+	}
+
+	var err error
+	i.f, err = newFreshConfig(deps.Config.GetString("otel.extension_url"), getter)
+	if err != nil {
+		// panic?
+		return provides{}
+	}
+
+	i.InventoryPayload = util.CreateInventoryPayload(deps.Config, deps.Log, deps.Serializer, i.getPayload, "otel.json")
+
+	if i.Enabled {
+
+		// TODO: if there's an update on the OTel side we currently will not be
+		//       notified. Is this a problem? Runtime changes are expected to be
+		//       triggered by FA, so maybe this is OK.
+		//
+		// We want to be notified when the configuration is updated
+		deps.Config.OnUpdate(func(_ string, _, _ any) { i.Refresh() })
```

Comment:
I don't think this is needed as we don't send the core-agent configuration here, no ?